### PR TITLE
Support consuming json stream and ndjson in REST Client Reactive

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
refs #21442

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
CC @Sgitario

**Review by geoand — APPROVED** [review]

**Review by Sgitario — COMMENTED** [review]
lgtm! only a few nitpicker comments

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/MultiInvoker.java [hunk 1]
```diff
@@ -161,17 +170,70 @@ public void onCancel(Runnable onCancel) {
...
-                    R item = restClientRequestContext.readEntity(in, responseType, response.getMediaType(),
-                            response.getMetadata());
-                    multiRequest.emitter.emit(item);
+                    byte[] bytes = buffer.getBytes();
+                    MediaType mediaType = response.getMediaType();
+
+                    if (isNewlineDelimited) {
+                        String charset = mediaType.getParameters().get(MediaType.CHARSET_PARAMETER);
+                        charset = charset == null ? "UTF-8" : charset;
+                        byte[] separator = "\n".getBytes(charset);
```
[the remark this answers is not in the document]
Comment (michalszynkiewicz) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/MultiInvoker.java:179]: Someone can use a different back-end than Quarkus. If someone changes `\n` there to `\n\r` we still want `\n` here, I think.

File: independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/providers/serialisers/AbstractJsonMessageBodyReader.java [hunk 2]
```diff
@@ -21,7 +21,8 @@ protected boolean isReadable(MediaType mediaType, Class<?> type) {
         }
         String subtype = mediaType.getSubtype();
         boolean isApplicationMediaType = "application".equals(mediaType.getType());
-        return (isApplicationMediaType && "json".equalsIgnoreCase(subtype) || subtype.endsWith("+json"))
+        return (isApplicationMediaType && "json".equalsIgnoreCase(subtype) || subtype.endsWith("+json")
```
[the remark this answers is not in the document]
Comment (michalszynkiewicz) [reply independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/providers/serialisers/AbstractJsonMessageBodyReader.java:24]: This should be a bit faster than checking compatibility on types...

**Pull request merged**